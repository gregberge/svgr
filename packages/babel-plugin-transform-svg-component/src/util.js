export const getProps = ({ types: t }, opts) => {
  const props = []

  if (opts.ref) {
    props.push(
      t.objectProperty(
        t.identifier('svgRef'),
        t.identifier('svgRef'),
        false,
        true,
      ),
    )
  }

  if (opts.titleProp) {
    props.push(
      t.objectProperty(
        t.identifier('title'),
        t.identifier('title'),
        false,
        true,
      ),
    )
  }

  if (opts.expandProps) {
    props.push(t.restElement(t.identifier('props')))
  }

  if (props.length === 0) {
    return null
  }

  if (props.length === 1 && opts.expandProps) {
    return t.identifier('props')
  }

  return t.objectPattern(props)
}

export const getImport = ({ types: t }, opts) => {
  const importDeclarations = [
    t.importDeclaration(
      [t.importDefaultSpecifier(t.identifier('React'))],
      t.stringLiteral('react'),
    ),
  ]

  if (opts.native) {
    importDeclarations.push(
      t.importDeclaration(
        [t.importDefaultSpecifier(t.identifier('Svg'))],
        t.stringLiteral('react-native-svg'),
      ),
    )
  }

  return importDeclarations
}

export const getExport = ({ template }, opts) => {
  let result = ''
  let exportName = opts.state.componentName

  if (opts.ref) {
    exportName = 'ForwardRef'
    result += `const ForwardRef = React.forwardRef((props, ref) => <${
      opts.state.componentName
    } svgRef={ref} {...props} />)\n\n`
  }

  if (opts.state.caller && opts.state.caller.previousExport) {
    result += `${opts.state.caller.previousExport}\n`
    result += `export { ${exportName} as ReactComponent }`
    return template.ast(result, {
      plugins: ['jsx'],
    })
  }

  result += `export default ${exportName}`
  return template.ast(result, {
    plugins: ['jsx'],
  })
}
