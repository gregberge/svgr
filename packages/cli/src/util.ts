/* eslint-disable no-console */
import { promises as fs } from 'fs'
import { red } from 'chalk'
import { transform, Config, State } from '@svgr/core'
import svgo from '@svgr/plugin-svgo'
import jsx from '@svgr/plugin-jsx'
import prettier from '@svgr/plugin-prettier'
// @ts-ignore
import camelCase from 'camelcase'
// @ts-ignore
import dashify from 'dashify'

export function transformFilename(
  filename: string,
  filenameCase: string,
): string {
  switch (filenameCase) {
    case 'kebab':
      return dashify(filename.replace(/_/g, '-'), { condense: true })
    case 'camel':
      return camelCase(filename)
    case 'pascal':
      return camelCase(filename, { pascalCase: true })
    default:
      throw new Error(`Unknown --filename-case ${filenameCase}`)
  }
}

export const convert = (
  code: string,
  config: Config,
  state: Partial<State>,
): string => {
  return transform.sync(code, config, {
    ...state,
    caller: {
      name: '@svgr/cli',
      defaultPlugins: [svgo, jsx, prettier],
    },
  })
}

export const convertFile = async (
  filePath: string,
  config: Config = {},
): Promise<string> => {
  const code = await fs.readFile(filePath, 'utf-8')
  return convert(code, config, { filePath })
}

export const exitError = (error: string): never => {
  console.error(red(error))
  process.exit(1)
}

export const politeWrite = (data: string, silent?: boolean): void => {
  if (!silent) {
    process.stdout.write(data)
  }
}

export const formatExportName = (name: string): string => {
  if (/[-]/g.test(name) && /^\d/.test(name)) {
    return `Svg${camelCase(name, { pascalCase: true })}`
  }

  if (/^\d/.test(name)) {
    return `Svg${name}`
  }

  return camelCase(name, { pascalCase: true })
}
