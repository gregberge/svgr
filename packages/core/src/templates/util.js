export const getProps = config => {
  const props = []
  if (config.ref) props.push('svgRef')
  if (config.titleProp) props.push('title')
  if (config.expandProps) props.push('...props')

  if (props.length === 0) return '()'
  if (props.length === 1 && config.expandProps) return 'props'

  return `({ ${props.join(', ')} })`
}
