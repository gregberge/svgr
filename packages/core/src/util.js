import path from 'path'
import { upperFirst, camelCase } from 'lodash'

export const pascalCase = str => upperFirst(camelCase(str))

export function getComponentName(state) {
  if (!state.filePath) return 'SvgComponent'
  const componentName = pascalCase(path.parse(state.filePath).name)
  if (Number.isNaN(parseInt(componentName[0], 10))) return componentName
  return `Svg${componentName}`
}

export function expandState(state) {
  return {
    componentName: state.componentName || getComponentName(state),
    ...state,
  }
}
