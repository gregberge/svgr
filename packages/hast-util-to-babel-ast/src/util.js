/**
 * Determines if the specified string consists entirely of numeric characters.
 *
 * @param {*} [value]
 * @returns {boolean}
 */
export function isNumeric(value) {
  return !Number.isNaN(value - parseFloat(value))
}

/**
 * Convert a hyphenated string to camelCase.
 *
 * @param {string} string
 * @returns {string}
 */
export function hyphenToCamelCase(string) {
  return string.replace(/-(.)/g, (match, chr) => chr.toUpperCase())
}

/**
 * Trim the specified substring off the string. If the string does not end
 * with the specified substring, this is a no-op.
 *
 * @param {string} haystack String to search in
 * @param {string} needle   String to search for
 * @return {string}
 */
export function trimEnd(haystack, needle) {
  return haystack.endsWith(needle)
    ? haystack.slice(0, -needle.length)
    : haystack
}
