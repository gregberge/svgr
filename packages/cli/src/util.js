/* eslint-disable no-console */
import fs from 'fs'
import chalk from 'chalk'
import util from 'util'
import convert, { resolveConfig } from '@svgr/core'

export const readFile = util.promisify(fs.readFile)
export const stat = util.promisify(fs.stat)

export async function convertFile(filePath, { config, ...options } = {}) {
  const code = await readFile(filePath, 'utf-8')
  const rcConfig = await resolveConfig(filePath, config)
  return convert(code, { ...rcConfig, ...options }, { filePath })
}

export function exitError(error) {
  console.error(chalk.red(error))
  process.exit(1)
}
