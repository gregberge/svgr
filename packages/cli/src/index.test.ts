import { promises as fs } from 'fs'
import * as path from 'path'
import { exec as execCb } from 'child_process'
import { promisify } from 'util'
// @ts-ignore
import del from 'del'

const exec = promisify(execCb)

const svgr = path.join(__dirname, '../bin/svgr')

describe('cli', () => {
  const cli = async (args: string) => {
    const { stdout } = await exec(`${svgr} ${args}`)
    return stdout
  }

  it('should work with a simple file', async () => {
    const result = await cli('__fixtures__/simple/file.svg')
    expect(result).toMatchSnapshot()
  })

  it('should not work with a directory without --out-dir option', async () => {
    expect.assertions(1)
    try {
      await cli('__fixtures__/nesting')
    } catch (error: any) {
      expect(error.message).toMatch(
        'Directory are not supported without `--out-dir` option instead',
      )
    }
  })

  it('should not work with several files without destination', async () => {
    expect.assertions(1)
    try {
      await cli('__fixtures__/simple/file.svg __fixtures__/nesting/one.svg')
    } catch (error: any) {
      expect(error.message).toMatch(
        'Please specify only one filename or use `--out-dir` option',
      )
    }
  })

  it('should work with stdin', async () => {
    const result = await cli('< __fixtures__/simple/file.svg')
    expect(result).toMatchSnapshot()
  })

  it('should support stdin filepath', async () => {
    const result = await cli(
      '--stdin-filepath __fixtures__/simple/file.svg < __fixtures__/simple/file.svg',
    )
    expect(result).toMatchSnapshot()
  })

  it('should transform a whole directory and output relative destination paths', async () => {
    const result = await cli('--out-dir __fixtures_build__/whole __fixtures__')
    const sorted = result
      .split(/\n/)
      .sort()
      .map((x) => x.toLowerCase())
      .join('\n')
    expect(sorted).toMatchSnapshot()
  })

  it('should transform a whole directory with --typescript', async () => {
    const result = await cli(
      '--typescript --out-dir __fixtures_build__/whole __fixtures__',
    )
    const sorted = result
      .split(/\n/)
      .sort()
      .map((x) => x.toLowerCase())
      .join('\n')
    expect(sorted).toMatchSnapshot()
  })

  it('should suppress output when transforming a directory with a --silent option', async () => {
    const result = await cli(
      '--silent --out-dir __fixtures_build__/whole __fixtures__',
    )
    const sorted = result.split(/\n/).sort().join('\n')
    expect(sorted).toMatchSnapshot()
  })

  it('should support --prettier-config as json', async () => {
    const result = await cli(
      `--prettier-config '{"tabWidth": 5}' __fixtures__/simple/file.svg`,
    )
    expect(result).toMatchSnapshot()
  })

  it('should support --prettier-config as file', async () => {
    const result = await cli(
      `--prettier-config __fixtures__/withPrettierRc/.prettierrc __fixtures__/simple/file.svg`,
    )
    expect(result).toMatchSnapshot()
  })

  it('should support --svgo-config as json', async () => {
    const result = await cli(
      `--svgo-config '{"plugins":[{"name":"preset-default","params":{"overrides":{"removeTitle":false}}}]}' __fixtures__/simple/file.svg`,
    )
    expect(result).toMatchSnapshot()
  })

  it('should support --svgo-config as file', async () => {
    const result = await cli(
      `--svgo-config __fixtures__/withSvgoConfig/svgo.config.js __fixtures__/simple/file.svg`,
    )
    expect(result).toMatchSnapshot()
  })

  it.each([
    ['--no-dimensions'],
    ['--jsx-runtime classic-preact'],
    ['--jsx-runtime automatic'],
    ['--expand-props none'],
    ['--expand-props start'],
    ['--icon'],
    ['--icon 24'],
    ['--icon 2em'],
    ['--native'],
    ['--native --icon'],
    ['--native --expand-props none'],
    ['--native --ref'],
    ['--ref'],
    ['--replace-attr-values "#063855=currentColor"'],
    [`--svg-props "hidden={true},id=hello"`],
    ['--no-svgo'],
    ['--no-prettier'],
    ['--title-prop'],
    ['--typescript'],
    ['--typescript --ref'],
    ['--typescript --ref --title-prop'],
  ])(
    'should support various args',
    async (args) => {
      const result = await cli(`${args} -- __fixtures__/simple/file.svg`)
      expect(result).toMatchSnapshot(args)
    },
    10000,
  )

  it.each([
    [0, ''],
    [1, '--filename-case=camel'],
    [2, '--filename-case=pascal'],
    [3, '--filename-case=kebab'],
  ])(
    'should support different filename cases with directory output',
    async (index, args) => {
      const inDir = '__fixtures__/cased'
      const outDir = `__fixtures_build__/filename-case-${index}`
      await del(outDir)
      await cli(`${args} ${inDir} --out-dir=${outDir}`)
      expect(await fs.readdir(outDir)).toMatchSnapshot(args)
    },
    10000,
  )

  it('should support custom file extension', async () => {
    const inDir = '__fixtures__/simple'
    const outDir = '__fixtures_build__/ext'
    await del(outDir)
    await cli(`--ext=ts ${inDir} --out-dir=${outDir}`)
    expect(await fs.readdir(outDir)).toMatchSnapshot()
  })

  it('should support "--ignore-existing"', async () => {
    const inDir = '__fixtures__/simple'
    const outDir = '__fixtures__/simple-existing'
    await cli(`${inDir} --out-dir=${outDir} --ignore-existing`)
    const content = await fs.readFile(path.join(outDir, 'File.js'), 'utf-8')
    expect(content).toBe('// nothing')
  })

  it('should not override config with cli defaults', async () => {
    const result = await cli(
      '__fixtures__/simple/file.svg --config-file=__fixtures__/overrides.config.js',
    )
    expect(result).toMatchSnapshot()
  })

  it('should add Svg prefix to index.js exports staring with number', async () => {
    const inDir = '__fixtures__/numeric'
    const outDir = `__fixtures_build__/prefix-exports`
    await del(outDir)
    await cli(`${inDir} --out-dir=${outDir}`)
    const content = await fs.readFile(path.join(outDir, 'index.js'), 'utf-8')
    expect(content).toMatchSnapshot()
  })

  it('should support custom index.js with directory output', async () => {
    const inDir = '__fixtures__/simple'
    const outDir = `__fixtures_build__/custom-index`
    await del(outDir)
    await cli(
      `${inDir} --out-dir=${outDir} --config-file=__fixtures__/custom-index.config.js`,
    )
    const content = await fs.readFile(path.join(outDir, 'index.js'), 'utf-8')
    expect(content).toMatchSnapshot()
  })

  it('using typescript option, it creates index with `.ts` extension', async () => {
    const inDir = '__fixtures__/simple'
    const outDir = `__fixtures_build__/ts-index`
    await del(outDir)
    await cli(`${inDir} --out-dir=${outDir} --typescript`)
    const content = await fs.readFile(path.join(outDir, 'index.ts'), 'utf-8')
    expect(content).toMatchSnapshot()
  })

  it('should support --index-template in cli', async () => {
    const inDir = '__fixtures__/simple'
    const outDir = `__fixtures_build__/custom-index-arg`
    await del(outDir)
    await cli(
      `${inDir} --out-dir=${outDir} --index-template=__fixtures__/custom-index-template.js`,
    )
    const content = await fs.readFile(path.join(outDir, 'index.js'), 'utf-8')
    expect(content).toMatchSnapshot()
  })

  it('should support --no-index', async () => {
    const inDir = '__fixtures__/simple'
    const outDir = `__fixtures_build__/no-index-case`
    await del(outDir)
    await cli(`--no-index ${inDir} --out-dir=${outDir}`)
    expect(await fs.readdir(outDir)).toMatchSnapshot()
  })
})
