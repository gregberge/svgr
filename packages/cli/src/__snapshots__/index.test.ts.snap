// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli should add Svg prefix to index.js exports staring with number 1`] = `
"export { default as Svg2File } from './2File'
export { default as File } from './File'
"
`;

exports[`cli should not override config with cli defaults 1`] = `
"import * as React from 'react'

const SvgFile = () => (
  <svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\">
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support --index-template in cli 1`] = `
"export { File } from './File'
"
`;

exports[`cli should support --no-index 1`] = `
Array [
  "File.js",
]
`;

exports[`cli should support --prettier-config as file 1`] = `
"import * as React from 'react'

const SvgFile = (props) => (
     <svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\" {...props}>
          <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
     </svg>
)

export default SvgFile

"
`;

exports[`cli should support --prettier-config as json 1`] = `
"import * as React from 'react'

const SvgFile = (props) => (
     <svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\" {...props}>
          <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
     </svg>
)

export default SvgFile

"
`;

exports[`cli should support --svgo-config as file 1`] = `
"import * as React from 'react'

const SvgFile = (props) => (
  <svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\" {...props}>
    <title>{'Rectangle 5'}</title>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support --svgo-config as json 1`] = `
"import * as React from 'react'

const SvgFile = (props) => (
  <svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\" {...props}>
    <title>{'Rectangle 5'}</title>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support custom file extension 1`] = `
Array [
  "File.ts",
  "index.ts",
]
`;

exports[`cli should support custom index.js with directory output 1`] = `
"export { File } from './File'
"
`;

exports[`cli should support different filename cases with directory output 1`] = `
Array [
  "CamelCase.js",
  "KebabCase.js",
  "MultipleDashes.js",
  "PascalCase.js",
  "index.js",
]
`;

exports[`cli should support different filename cases with directory output: --filename-case=camel 1`] = `
Array [
  "camelCase.js",
  "index.js",
  "kebabCase.js",
  "multipleDashes.js",
  "pascalCase.js",
]
`;

exports[`cli should support different filename cases with directory output: --filename-case=kebab 1`] = `
Array [
  "camel-case.js",
  "index.js",
  "kebab-case.js",
  "multiple-dashes.js",
  "pascal-case.js",
]
`;

exports[`cli should support different filename cases with directory output: --filename-case=pascal 1`] = `
Array [
  "CamelCase.js",
  "KebabCase.js",
  "MultipleDashes.js",
  "PascalCase.js",
  "index.js",
]
`;

exports[`cli should support stdin filepath 1`] = `
"import * as React from 'react'

const SvgFile = (props) => (
  <svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\" {...props}>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --expand-props none 1`] = `
"import * as React from 'react'

const SvgFile = () => (
  <svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\">
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --expand-props start 1`] = `
"import * as React from 'react'

const SvgFile = (props) => (
  <svg {...props} width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\">
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --icon 1`] = `
"import * as React from 'react'

const SvgFile = (props) => (
  <svg
    width=\\"1em\\"
    height=\\"1em\\"
    viewBox=\\"0 0 48 1\\"
    xmlns=\\"http://www.w3.org/2000/svg\\"
    {...props}
  >
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --icon 2em 1`] = `
"import * as React from 'react'

const SvgFile = (props) => (
  <svg
    width=\\"2em\\"
    height=\\"2em\\"
    viewBox=\\"0 0 48 1\\"
    xmlns=\\"http://www.w3.org/2000/svg\\"
    {...props}
  >
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --icon 24 1`] = `
"import * as React from 'react'

const SvgFile = (props) => (
  <svg
    width={24}
    height={24}
    viewBox=\\"0 0 48 1\\"
    xmlns=\\"http://www.w3.org/2000/svg\\"
    {...props}
  >
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --jsx-runtime automatic 1`] = `
"const SvgFile = (props) => (
  <svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\" {...props}>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --jsx-runtime classic-preact 1`] = `
"import { h } from 'preact'

const SvgFile = (props) => (
  <svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\" {...props}>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --native --expand-props none 1`] = `
"import * as React from 'react'
import Svg, { Path } from 'react-native-svg'

const SvgFile = () => (
  <Svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\">
    <Path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </Svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --native --icon 1`] = `
"import * as React from 'react'
import Svg, { Path } from 'react-native-svg'

const SvgFile = (props) => (
  <Svg
    width={24}
    height={24}
    viewBox=\\"0 0 48 1\\"
    xmlns=\\"http://www.w3.org/2000/svg\\"
    {...props}
  >
    <Path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </Svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --native --ref 1`] = `
"import * as React from 'react'
import Svg, { Path } from 'react-native-svg'
import { forwardRef } from 'react'

const SvgFile = (props, ref) => (
  <Svg
    width={48}
    height={1}
    xmlns=\\"http://www.w3.org/2000/svg\\"
    ref={ref}
    {...props}
  >
    <Path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </Svg>
)

const ForwardRef = forwardRef(SvgFile)
export default ForwardRef

"
`;

exports[`cli should support various args: --native 1`] = `
"import * as React from 'react'
import Svg, { Path } from 'react-native-svg'

const SvgFile = (props) => (
  <Svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\" {...props}>
    <Path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </Svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --no-dimensions 1`] = `
"import * as React from 'react'

const SvgFile = (props) => (
  <svg viewBox=\\"0 0 48 1\\" xmlns=\\"http://www.w3.org/2000/svg\\" {...props}>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --no-prettier 1`] = `
"import * as React from \\"react\\";

const SvgFile = props => <svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\" {...props}><path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" /></svg>;

export default SvgFile;
"
`;

exports[`cli should support various args: --no-svgo 1`] = `
"import * as React from 'react'

const SvgFile = (props) => (
  <svg
    width=\\"48px\\"
    height=\\"1px\\"
    viewBox=\\"0 0 48 1\\"
    xmlns=\\"http://www.w3.org/2000/svg\\"
    xmlnsXlink=\\"http://www.w3.org/1999/xlink\\"
    {...props}
  >
    <title>{'Rectangle 5'}</title>
    <desc>{'Created with Sketch.'}</desc>
    <defs />
    <g id=\\"Page-1\\" stroke=\\"none\\" strokeWidth={1} fill=\\"none\\" fillRule=\\"evenodd\\">
      <g
        id=\\"19-Separator\\"
        transform=\\"translate(-129.000000, -156.000000)\\"
        fill=\\"#063855\\"
      >
        <g id=\\"Controls/Settings\\" transform=\\"translate(80.000000, 0.000000)\\">
          <g id=\\"Content\\" transform=\\"translate(0.000000, 64.000000)\\">
            <g id=\\"Group\\" transform=\\"translate(24.000000, 56.000000)\\">
              <g id=\\"Group-2\\">
                <rect id=\\"Rectangle-5\\" x={25} y={36} width={48} height={1} />
              </g>
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --ref 1`] = `
"import * as React from 'react'
import { forwardRef } from 'react'

const SvgFile = (props, ref) => (
  <svg
    width={48}
    height={1}
    xmlns=\\"http://www.w3.org/2000/svg\\"
    ref={ref}
    {...props}
  >
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

const ForwardRef = forwardRef(SvgFile)
export default ForwardRef

"
`;

exports[`cli should support various args: --replace-attr-values "#063855=currentColor" 1`] = `
"import * as React from 'react'

const SvgFile = (props) => (
  <svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\" {...props}>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"currentColor\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --svg-props "hidden={true},id=hello" 1`] = `
"import * as React from 'react'

const SvgFile = (props) => (
  <svg
    width={48}
    height={1}
    xmlns=\\"http://www.w3.org/2000/svg\\"
    hidden={true}
    id=\\"hello\\"
    {...props}
  >
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --title-prop 1`] = `
"import * as React from 'react'

const SvgFile = ({ title, titleId, ...props }) => (
  <svg
    width={48}
    height={1}
    xmlns=\\"http://www.w3.org/2000/svg\\"
    aria-labelledby={titleId}
    {...props}
  >
    {title ? <title id={titleId}>{title}</title> : null}
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should support various args: --typescript --ref --title-prop 1`] = `
"import * as React from 'react'
import { SVGProps, Ref, forwardRef } from 'react'
interface SVGRProps {
  title?: string;
  titleId?: string;
}

const SvgFile = (
  { title, titleId, ...props }: SVGProps<SVGSVGElement> & SVGRProps,
  ref: Ref<SVGSVGElement>,
) => (
  <svg
    width={48}
    height={1}
    xmlns=\\"http://www.w3.org/2000/svg\\"
    ref={ref}
    aria-labelledby={titleId}
    {...props}
  >
    {title ? <title id={titleId}>{title}</title> : null}
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

const ForwardRef = forwardRef(SvgFile)
export default ForwardRef

"
`;

exports[`cli should support various args: --typescript --ref 1`] = `
"import * as React from 'react'
import { SVGProps, Ref, forwardRef } from 'react'

const SvgFile = (props: SVGProps<SVGSVGElement>, ref: Ref<SVGSVGElement>) => (
  <svg
    width={48}
    height={1}
    xmlns=\\"http://www.w3.org/2000/svg\\"
    ref={ref}
    {...props}
  >
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

const ForwardRef = forwardRef(SvgFile)
export default ForwardRef

"
`;

exports[`cli should support various args: --typescript 1`] = `
"import * as React from 'react'
import { SVGProps } from 'react'

const SvgFile = (props: SVGProps<SVGSVGElement>) => (
  <svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\" {...props}>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should suppress output when transforming a directory with a --silent option 1`] = `""`;

exports[`cli should transform a whole directory and output relative destination paths 1`] = `
"
__fixtures__/cased/pascalcase.svg -> __fixtures_build__/whole/cased/pascalcase.js
__fixtures__/cased/camelcase.svg -> __fixtures_build__/whole/cased/camelcase.js
__fixtures__/cased/kebab-case.svg -> __fixtures_build__/whole/cased/kebabcase.js
__fixtures__/cased/multiple---dashes.svg -> __fixtures_build__/whole/cased/multipledashes.js
__fixtures__/complex/skype.svg -> __fixtures_build__/whole/complex/skype.js
__fixtures__/complex/telegram.svg -> __fixtures_build__/whole/complex/telegram.js
__fixtures__/nesting/a/c/three.svg -> __fixtures_build__/whole/nesting/a/c/three.js
__fixtures__/nesting/a/two.svg -> __fixtures_build__/whole/nesting/a/two.js
__fixtures__/nesting/one.svg -> __fixtures_build__/whole/nesting/one.js
__fixtures__/numeric/2.file.svg -> __fixtures_build__/whole/numeric/2file.js
__fixtures__/numeric/file.svg -> __fixtures_build__/whole/numeric/file.js
__fixtures__/simple/file.svg -> __fixtures_build__/whole/simple/file.js
__fixtures__/withprettierrc/file.svg -> __fixtures_build__/whole/withprettierrc/file.js
__fixtures__/withsvgoconfig/file.svg -> __fixtures_build__/whole/withsvgoconfig/file.js
__fixtures__/withsvgrrc/file.svg -> __fixtures_build__/whole/withsvgrrc/file.js"
`;

exports[`cli should transform a whole directory with --typescript 1`] = `
"
__fixtures__/cased/pascalcase.svg -> __fixtures_build__/whole/cased/pascalcase.tsx
__fixtures__/cased/camelcase.svg -> __fixtures_build__/whole/cased/camelcase.tsx
__fixtures__/cased/kebab-case.svg -> __fixtures_build__/whole/cased/kebabcase.tsx
__fixtures__/cased/multiple---dashes.svg -> __fixtures_build__/whole/cased/multipledashes.tsx
__fixtures__/complex/skype.svg -> __fixtures_build__/whole/complex/skype.tsx
__fixtures__/complex/telegram.svg -> __fixtures_build__/whole/complex/telegram.tsx
__fixtures__/nesting/a/c/three.svg -> __fixtures_build__/whole/nesting/a/c/three.tsx
__fixtures__/nesting/a/two.svg -> __fixtures_build__/whole/nesting/a/two.tsx
__fixtures__/nesting/one.svg -> __fixtures_build__/whole/nesting/one.tsx
__fixtures__/numeric/2.file.svg -> __fixtures_build__/whole/numeric/2file.tsx
__fixtures__/numeric/file.svg -> __fixtures_build__/whole/numeric/file.tsx
__fixtures__/simple/file.svg -> __fixtures_build__/whole/simple/file.tsx
__fixtures__/withprettierrc/file.svg -> __fixtures_build__/whole/withprettierrc/file.tsx
__fixtures__/withsvgoconfig/file.svg -> __fixtures_build__/whole/withsvgoconfig/file.tsx
__fixtures__/withsvgrrc/file.svg -> __fixtures_build__/whole/withsvgrrc/file.tsx"
`;

exports[`cli should work with a simple file 1`] = `
"import * as React from 'react'

const SvgFile = (props) => (
  <svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\" {...props}>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgFile

"
`;

exports[`cli should work with stdin 1`] = `
"import * as React from 'react'

const SvgComponent = (props) => (
  <svg width={48} height={1} xmlns=\\"http://www.w3.org/2000/svg\\" {...props}>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default SvgComponent

"
`;

exports[`cli using typescript option, it creates index with \`.ts\` extension 1`] = `
"export { default as File } from './File'
"
`;
